### PR TITLE
Feat/fly 2542

### PR DESCRIPTION
**Pull request by Hakob23**
## What

Implement the frozen peg-out side of `FlyoverConfigurations`: active config reads, satoshi-floored `calculatePegOutFee`, confirmation-tier lookup, and time-locked `queuePegOutChange` / `applyPegOutChange`, with peg-out storage in a separate ERC-7201 namespace and one-shot `initializePegOut`. Wire provisional regtest seeds and update the test mock. No interface changes.

## Why

Peg-out protocol parameters need the same on-chain, time-locked source as peg-in so SDK/LPS/escrow all read one set of numbers. This PR replaces the `PegOutNotImplemented` stubs left by FLY-2541 without touching the frozen ABI.

## Type of Change

- [ ] Bug fix (non-breaking change)
- [x] New feature (non-breaking change)
- [ ] Breaking change
- [ ] Refactor (no intended behavior change)
- [ ] Security hardening
- [ ] Tests only
- [ ] Build/CI/Tooling
- [ ] Documentation

## Affected Areas

- [ ] PegIn flow (`registerPegIn`, `callForUser`, quote validation)
- [x] PegOut flow (`depositPegout`, `refundPegOut`, `refundUserPegOut`)
- [ ] Liquidity Provider lifecycle (register/update/status/resign/collateral/withdraw)
- [ ] Pause/operational controls
- [ ] Quote hashing/signature logic
- [ ] Bitcoin tx / PMT / merkle validation
- [ ] Deployment or upgrade scripts / Makefile targets
- [ ] CI workflows (`ci`, `fuzz`, `formal`, `slither`, `codeql`)
- [ ] Docs

## Risk & Security Review

- [x] Access control and authorization paths reviewed
- [ ] Reentrancy and external call paths reviewed
- [x] Storage layout / upgrade safety reviewed (if upgradeable code changed)
- [x] Time/block/confirmation assumptions reviewed
- [ ] BTC tx output/index assumptions reviewed (if pegout validation touched)
- [x] No secrets or sensitive values added

## Test Plan

List what you ran locally and the outcome:

- [ ] `npm run lint:sol`
- [ ] `npm run lint:ts`
- [ ] `npm run compile`
- [ ] `npm test`
- [ ] `npm run test:fuzz` (if logic/state transitions changed)
- [ ] `npm run test:invariant` (if invariant-sensitive logic changed)
- [ ] `npm run test:formal` (if formal-verification-sensitive logic changed)
- [ ] `npm run test:integration` (if integration paths changed)
- [ ] `npm run test:e2e` (if deployment/ownership flows changed)

Additional notes on test coverage, edge cases, and any tests intentionally skipped:

No new unit tests in this PR. Relies on existing configurations/pegin suites staying green and CI. Escrow snapshot economics will be covered when PegOutEscrow lands.

## Deployment & Ops Impact

- [ ] No deployment impact
- [x] Requires deployment
- [x] Requires upgrade
- [ ] Env/config changes required (`.env`, network RPCs, verification, etc.)
- [ ] Runbook/update instructions added to PR description

After peg-in `initialize`, admin must call `initializePegOut(config, min, max)` once to seed peg-out storage. Shared timelock delay is unchanged.

## Related Issues

- Jira: [FLY-2542](https://rsklabs.atlassian.net/browse/FLY-2542)
- Depends on: feat/FLY-2541 (frozen peg-out ABI + stubs)
- GitHub Issue: #(if applicable)

## Reviewer Notes

- Base this PR on `feat/FLY-2541` so interface edits stay out of this PR. To be merged with v3.0.0 once approved and after this [PR](https://github.com/rsksmart/liquidity-bridge-contract/pull/523) has been merged
- Peg-out lives under ERC-7201 `rsk.flyover.FlyoverConfigurations.pegOut` so peg-in slots are untouched.
- `initializePegOut` is intentionally separate from `initialize` (see TODO on that function).
- PoC fields `gasFee` / `depositConfirmations` are intentionally omitted (frozen 12-field struct).

## Screenshots / Logs (if applicable)

N/A

### CODE REVIEW

**Review by Freshenext — APPROVED** [review]

File: src/FlyoverConfigurations.sol
Comment (Luisfc68) [inline src/FlyoverConfigurations.sol:1]: If you decided not to reuse `PegConfiguration` then I suggest to rename it to `PegInConfiguration`

File: src/FlyoverConfigurations.sol [hunk 1]
```diff
@@ -130,6 +160,27 @@ contract FlyoverConfigurations is
...
+    /// unchanged, and so an already-initialized proxy can seed the peg-out ERC-7201 namespace
+    /// without re-running `initialize`.
+    // solhint-disable-next-line comprehensive-interface
+    function initializePegOut(
+        PegOutConfiguration calldata pegOutConfig,
+        PegOutConfiguration calldata pegOutMin,
+        PegOutConfiguration calldata pegOutMax
+    ) external onlyRole(DEFAULT_ADMIN_ROLE) {
+        PegOutConfigurationsStorage storage pegOut = _getPegOutStorage();
+        if (pegOut.initialized) revert PegOutAlreadyInitialized();
```
Comment (Luisfc68) [inline src/FlyoverConfigurations.sol:205]: use the initializer modifier so we don't have to maintain this ourselves and also to have access to the reinitializers later

File: src/FlyoverConfigurations.sol [hunk 2]
```diff
@@ -271,6 +380,21 @@ contract FlyoverConfigurations is
         return tiers[length - 1].confirmations;
     }
 
+    function _requiredPegOutConfirmations(PegOutConfiguration storage config, uint256 amount)
```
Comment (Luisfc68) [inline src/FlyoverConfigurations.sol:502]: can't this reuse the pegin function for the same purpose?

**Review by Luisfc68 — CHANGES_REQUESTED** [review]

File: src/FlyoverConfigurations.sol (hunk 1 above)
Comment (Hakob23) [reply src/FlyoverConfigurations.sol]: Fixes are applied in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

**Comment by Hakob23** [comment]
Fixes are applied in https://github.com/rsksmart/liquidity-bridge-contract/pull/549

File: src/FlyoverConfigurations.sol (hunk 2 above)
Comment (Hakob23) [reply src/FlyoverConfigurations.sol]: yes, fixed in https://github.com/rsksmart/liquidity-bridge-contract/pull/549
